src/Features/Onboarding.php

```diff
@@ -418,24 +419,109 @@ public static function update_help_tab() {
+			if ( class_exists( 'Jetpack' ) ) {
+				$help_tab['content'] .= '<p>' . __( 'Quickly access the Jetpack connection flow in Calypso.', 'woocommerce-admin' ) . '</p>';
+				$help_tab['content'] .= '<p><a href="' . wc_admin_url( '&test_wc_jetpack_connect=1' ) . '" class="button button-primary">' . __( 'Connect', 'woocommerce-admin' ) . '</a></p>';
+			}
+
+			$help_tab['content'] .= '<p>' . __( 'Quickly access the WooCommerce.com connection flow in Calypso.', 'woocommerce-admin' ) . '</p>';
+			$help_tab['content'] .= '<p><a href="' . wc_admin_url( '&test_wc_helper_connect=1' ) . '" class="button button-primary">' . __( 'Connect', 'woocommerce-admin' ) . '</a></p>';
+
+
 			$screen->add_help_tab( $help_tab );
 		}
+	}
+
+	/**
+	 * Allows quick access to testing the calypso parts of onboarding.
+	 */
+	public static function calypso_tests() {
+		$calypso_env = defined( 'WOOCOMMERCE_CALYPSO_ENVIRONMENT' ) && in_array( WOOCOMMERCE_CALYPSO_ENVIRONMENT, array( 'development', 'wpcalypso', 'horizon', 'stage' ) ) ? WOOCOMMERCE_CALYPSO_ENVIRONMENT : 'production';
+
+		if ( Loader::is_admin_page() && class_exists( 'Jetpack' ) && isset( $_GET['test_wc_jetpack_connect'] ) && 1 === absint( $_GET['test_wc_jetpack_connect'] ) ) { // WPCS: CSRF ok.
+			$redirect_url = esc_url_raw(
+				add_query_arg(
+					array(
+						'page' => 'wc-admin',
+					),
+					admin_url( 'admin.php' )
+				)
+			);
+
+			$connect_url = \Jetpack::init()->build_connect_url( true, $redirect_url, 'woocommerce-setup-wizard' );
+			$connect_url = add_query_arg( array( 'calypso_env' => $calypso_env ), $connect_url );
+
+			wp_redirect( $connect_url );
+			exit;
+		}
+
+		if ( Loader::is_admin_page() && isset( $_GET['test_wc_helper_connect'] ) && 1 === absint( $_GET['test_wc_helper_connect'] ) ) { // WPCS: CSRF ok.
+			include_once WC_ABSPATH . 'includes/admin/helper/class-wc-helper-api.php';
 
-		$task_list_hidden = get_option( 'woocommerce_task_list_hidden', 'no' );
-		$task_list_tab    = array(
-			'id'      => 'woocommerce_task_list_tab',
-			'title'   => __( 'Task list', 'woocommerce-admin' ),
-			'content' => '<h2>' . __( 'Task list', 'woocommerce-admin' ) . '</h2>' .
-				'<p>' . __( 'If you need to enable or disable the task list, please click on the button below.', 'woocommerce-admin' ) . '</p>' .
-				( 'yes' === $task_list_hidden
-					? '<p><a href="' . wc_admin_url( '&reset_task_list=1' ) . '" class="button button-primary">' . __( 'Enable', 'woocommerce-admin' ) . '</a></p>'
-					: '<p><a href="' . wc_admin_url( '&reset_task_list=0' ) . '" class="button button-primary">' . __( 'Disable', 'woocommerce-admin' ) . '</a></p>'
+			$redirect_uri = wc_admin_url( '&task=connect&wccom-connected=1' );
+
+			$request = \WC_Helper_API::post(
+				'oauth/request_token',
+				array(
+					'body' => array(
+						'home_url'     => home_url(),
+						'redirect_uri' => $redirect_uri,
+					),
+				)
+			);
+
+			$code = wp_remote_retrieve_response_code( $request );
+			if ( 200 !== $code ) {
```

Comment:
Should we thrown an error when this happens?

Reply:
https://github.com/woocommerce/woocommerce-admin/pull/2863/commits/52ef43e6054181fd540f56f12d5f12d62a2cc2ba